tests/foreman/cli/test_leapp_client.py

```diff
@@ -0,0 +1,375 @@
+"""Tests for leapp upgrade of content hosts with Satellite
+
+:Requirement: leapp
+
+:CaseLevel: Integration
+
+:CaseComponent: LeappIntegration
+
+:Team: Rocket
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:CaseAutomation: Automated
+
+:Upstream: No
+"""
+import pytest
+from broker import Broker
+
+from robottelo.cli.activationkey import ActivationKey
+from robottelo.cli.repository import Repository
+from robottelo.config import settings
+from robottelo.constants import PRDS
+from robottelo.hosts import ContentHost
+from robottelo.logging import logger
+
+synced_repos = pytest.StashKey[dict]
+
+
+@pytest.fixture(scope='module')
+def module_stash(request):
+    """Module scoped stash for storing data between tests"""
+    # Please refer the documentation for more details on stash
+    # https://docs.pytest.org/en/latest/reference/reference.html#stash
+    request.node.stash[synced_repos] = {}
```

Comment:
Just curious, what's need of pytest stash here?

Reply:
There is one test in the test module that is parametrized to two upgrade paths. 7to8 and 8to9. If we do not have the stash implemented, the second upgrade path will fail to enable the RHEL 8.8 repository that is already enabled on target_sat.
Stash enables us to store the synced/enabled repos in it so we can skip their enablement and sync before we try to enable and/or sync them.

Reply:
@Gauravtalreja1 Do you mean, why we are using pytest stash if we are making test module parameterised?
stash is for storing data between tests, but here we are using single test module.